File: apple/Sources/ReactNativeWatchOS/Modifiers/MoveDisabled.swift
import SwiftUI

/// SwiftUI `.moveDisabled(_:)`. Prevents a row in an editable `List` /
/// `ForEach` from being reordered. Defaults to `true` when unspecified.
enum RNWMoveDisabledModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("moveDisabled") { view, params, _ in
            AnyView(view.moveDisabled(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/moveDisabled.ts
import { createModifier } from './createModifier';

/// SwiftUI `.moveDisabled(_:)`. Prevents a row in an editable `List` /
/// `ForEach` from being reordered (moved) in edit mode.
export function moveDisabled(value: boolean = true) {
  return createModifier('moveDisabled', { value });
}
